internal/commands/app/create_inputs.go

```diff
@@ -156,56 +166,117 @@ func (i *createInputs) resolveLocalPath(ui terminal.UI, wd string) (string, erro
 	return fullPath, nil
 }
 
-func (i *createInputs) resolveCluster(client atlas.Client, groupID string) (dataSourceCluster, error) {
+func (i *createInputs) resolveClusters(ui terminal.UI, client atlas.Client, groupID string) ([]dataSourceCluster, error) {
 	clusters, err := client.Clusters(groupID)
 	if err != nil {
-		return dataSourceCluster{}, err
+		return nil, err
 	}
-	var clusterName string
-	for _, cluster := range clusters {
-		if i.Cluster == cluster.Name {
-			clusterName = cluster.Name
-			break
-		}
+
+	existingClusters := map[string]interface{}{}
```

Comment:
oops, sorry I should have clarified this one better.  so the purpose of `struct{}` is to have the smallest allocations possible (useful for when you're only tracking unique key values).  so something like this:

```go
existingClusters := map[string]struct{}{} // another pattern you'll see is people using `bool` here instead
for _, c := range clusters {
  existingClusters[c.Name] = struct{}{} // but `struct{}` has just slightly a smaller memory footprint
}
// can check out for more info: https://gist.github.com/davecheney/3be245c92b61e5045f75
```

now full disclosure, i may be missing something here.  if we actually do need the value from this map at some point then this won't work (but considering you got away with `interface{}` here already I'm guessing that's not the case)